src/shared/components/common/pending_follow.tsx

```diff
@@ -0,0 +1,113 @@
+import { Component, InfernoNode } from "inferno";
+import {
+  ApproveCommunityPendingFollower,
+  MyUserInfo,
+  PendingFollow as PendingFollowView,
+} from "lemmy-js-client";
+import { I18NextService } from "../../services";
+import { PersonListing } from "../person/person-listing";
+import { Spinner } from "./icon";
+import { UserBadges } from "./user-badges";
+import { linkEvent } from "inferno";
+
+interface PendingFollowProps {
+  pending_follow: PendingFollowView;
+  myUserInfo: MyUserInfo | undefined;
+  onApproveFollower(form: ApproveCommunityPendingFollower): void;
+}
+
+interface PendingFollowState {
+  approveLoading: boolean;
+  denyLoading: boolean;
+}
+
+export class PendingFollow extends Component<
+  PendingFollowProps,
+  PendingFollowState
+> {
+  state: PendingFollowState = {
+    approveLoading: false,
+    denyLoading: false,
+  };
+
+  constructor(props: any, context: any) {
+    super(props, context);
+  }
+  componentWillReceiveProps(
+    nextProps: Readonly<{ children?: InfernoNode } & PendingFollowProps>,
+  ): void {
+    if (this.props !== nextProps) {
+      this.setState({
+        approveLoading: false,
+        denyLoading: false,
+      });
+    }
+  }
+
+  render() {
+    const p = this.props.pending_follow;
+    return (
+      <div className="d-flex flex-column">
```

Comment:
Use the bootstrap grid classes if you need to. `registration-application.tsx` does this.

Reply:
Seems fine like this.

Reply:
Okay changed.